### PR TITLE
feat(chat): add option to unpublish app, add apps filter for publish request modal, hide caret icon if publication resource types don't contain prompt or conversation (Issues #1940, #2006, #2031)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

add option to unpublish app
add apps filter for publish request modal
hide caret icon if publication resource types don't contain prompt or conversation

Issues:

- #1940 
- #2006 
- #2031 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2033.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ModelList.tsx [hunk 1]
```diff
@@ -319,7 +323,9 @@ export const ModelList = ({
   const [modalIsOpen, setModalIsOpen] = useState(false);
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
   const [currentEntity, setCurrentEntity] = useState<DialAIEntityModel>();
-  const [isPublishing, setIsPublishing] = useState(false);
+  const [publishAction, setPublishAction] = useState<
+    PublishActions | undefined
+  >(undefined);
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ModelList.tsx]:
```suggestion
  >();
```

File: apps/chat/src/components/Chat/ModelList.tsx [hunk 2]
```diff
@@ -342,16 +348,16 @@ export const ModelList = ({
...
-  const handlePublish = () => {
-    setIsPublishing(true);
-  };
+  const handlePublish = useCallback((action: PublishActions) => {
+    setPublishAction(action);
+  }, []);
 
   const handlePublishClose = () => {
-    setIsPublishing(false);
+    setPublishAction(undefined);
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ModelList.tsx:359]:
```suggestion
    setPublishAction();
```

Comment (Alexander-Kezik) [reply apps/chat/src/components/Chat/ModelList.tsx:359]: setState must receive 1 arg

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2033.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**